soc/xtensa/intel_adsp/ace/power.c

```diff
@@ -134,27 +133,37 @@ void dsp_restore_vector(void);
 
 void power_gate_entry(uint32_t core_id)
 {
-	struct lpsram_header *lpsheader =
-		(struct lpsram_header *) DT_REG_ADDR(DT_NODELABEL(sram1));
-
 	xthal_window_spill();
+	z_xtensa_cache_flush_inv_all();
```

Comment:
why don't you do this in the end of this function and avoid the following flushes ?

Reply:
I need to do cache flush/invalidation for all after window spill and before context saving. It will not work otherwise.